### PR TITLE
add deadline option to any function that contains getprotocol or depe…

### PR DESCRIPTION
**Pull request by psi29a**
…nds on function that does

This also includes NotMasterError

### CODE REVIEW

File: tests/test_replicaset.py [hunk 1]
```diff
@@ -144,18 +144,133 @@ def test_AutoReconnect(self):
...
+            conn = ConnectionPool(uri, initial_delay=3)
+
+            yield conn.db.coll.insert({'x': 42}, safe=True)
+
+            yield self.__mongod[0].stop()
+
+            while True:
+                try:
+                    yield conn.db.coll.find_one(timeout=2)
+                    raise Exception("TimeoutExceeded not raised!")
```
Comment (IlyaSkriblovsky) [inline tests/test_replicaset.py]: it's more clean to use `self.fail()` here

File: tests/test_replicaset.py [hunk 2]
```diff
@@ -144,18 +144,133 @@ def test_AutoReconnect(self):
...
+
+            yield conn.db.coll.insert({'x': 42}, safe=True)
+
+            yield self.__mongod[0].stop()
+
+            while True:
+                try:
+                    deadline = time() + 2
+                    yield conn.db.coll.find_one(deadline=deadline)
+                    raise Exception("DeadlineExceeded not raised!")
```
Comment (IlyaSkriblovsky) [inline tests/test_replicaset.py]: and here too

File: txmongo/collection.py [hunk 3]
```diff
@@ -541,7 +545,7 @@ def ensure_index(self, sort_fields, **kwargs):
         # keeping an index cache same way pymongo does
         return self.create_index(sort_fields, **kwargs)
 
-    def drop_index(self, index_identifier):
+    def drop_index(self, index_identifier, **kwargs):
```
Comment (IlyaSkriblovsky) [inline txmongo/collection.py]: What is the reason for `**kwargs` here?

**Comment by IlyaSkriblovsky** [comment]
Is this intended that `count`, `group`, `filemd5` and some others have only `deadline` argument, but not `timeout`?

File: tests/test_replicaset.py (hunk 1 above)
Comment (psi29a) [inline tests/test_replicaset.py]: Right, thanks!

File: txmongo/collection.py (hunk 3 above)
Comment (psi29a) [inline txmongo/collection.py]: Absolutely none, removed.

**Comment by psi29a** [comment]
I'm working through them, thanks for noticing!

**Comment by psi29a** [comment]
Did a rebase squash in prep for merge.

**Pull request merged**